Microsoft.Toolkit.Uwp.UI.Animations/Extensions/AnimationExtensions.Light.cs

```diff
@@ -0,0 +1,201 @@
+﻿// ******************************************************************
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Graphics.Canvas;
+using Microsoft.Graphics.Canvas.Effects;
+using Windows.UI.Composition;
+using Windows.UI.Composition.Effects;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Hosting;
+
+namespace Microsoft.Toolkit.Uwp.UI.Animations
+{
+    /// <summary>
+    /// Provides an extension which allows lighting.
+    /// </summary>
+    public static partial class AnimationExtensions
+    {
+        /// <summary>
+        /// Stores all the point lights along with the visuals that they are applied to.
+        /// This is to stop mulitplication of point lights on a single visual.
+        /// </summary>
+        private static Dictionary<Visual, PointLight> pointLights = new Dictionary<Visual, PointLight>();
+
+        /// <summary>
+        /// Gets a value indicating whether this instance is lighting supported.
+        /// </summary>
+        /// <value>
+        /// <c>true</c> if this instance is lighting supported; otherwise, <c>false</c>.
+        /// </value>
+        public static bool IsLightingSupported
+        {
+            get
+            {
+                bool lightingSupported = true;
+
+                if (!Windows.Foundation.Metadata.ApiInformation.IsMethodPresent("Windows.UI.Xaml.Hosting.ElementCompositionPreview", "SetElementChildVisual"))
+                {
+                    lightingSupported = false;
+                }
+
+                if (!Windows.Foundation.Metadata.ApiInformation.IsTypePresent("Windows.UI.Composition.CompositionSurfaceBrush"))
+                {
+                    lightingSupported = false;
+                }
+
+                return lightingSupported;
+            }
+        }
+
+        /// <summary>
+        /// Animates a point light and it's distance.
+        /// </summary>
+        /// <param name="associatedObject">The associated object.</param>
+        /// <param name="distance">The value.</param>
+        /// <param name="duration">The duration.</param>
+        /// <param name="delay">The delay.</param>
+        /// <returns>An animation set.</returns>
+        public static async Task<AnimationSet> LightAsync(
```

Comment:
The calls made in the composition APIs are async so I end up with a chain of async commands. This only breaks animation set if the developer ignores the warnings from the compiler telling them they need to make it awaitable. Is this something that can be added after the fact as well without any breaking changes?